tests/testthat/test-derive_param_computed.R

```diff
@@ -354,9 +361,119 @@ test_that("derive_param_computed Test 8: no new observations if a constant param
   )
 })
 
+## Test 9: compute multiple variables ----
+test_that("derive_param_computed Test 9: compute multiple variables, keep_nas", {
+  adlb_tbilialk <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVALC, ~ADTM,        ~ADTF,
+    "1",      "ALK2",   "Y",    "2021-05-13", NA_character_,
+    "1",      "TBILI2", "Y",    "2021-06-30", "D",
+    "2",      "ALK2",   "Y",    "2021-12-31", "M",
+    "2",      "TBILI2", "N",    "2021-11-11", NA_character_,
+    "3",      "ALK2",   "N",    "2021-04-03", NA_character_,
+    "3",      "TBILI2", "N",    "2021-04-04", NA_character_
+  ) %>%
+    mutate(ADTM = lubridate::ymd(ADTM))
+
+  expected <- tibble::tribble(
+    ~USUBJID, ~AVALC, ~ADTM,        ~ADTF,
+    "1",      "Y",    "2021-06-30", "D",
+    "2",      "N",    "2021-12-31", "M",
+    "3",      "N",    "2021-04-04", NA_character_
+  ) %>%
+    mutate(
+      ADTM = lubridate::ymd(ADTM),
+      PARAMCD = "TB2AK2",
+      PARAM = "TBILI > 2 times ULN and ALKPH <= 2 times ULN"
+    )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = derive_param_computed(
+      dataset_add = adlb_tbilialk,
+      by_vars = exprs(USUBJID),
+      parameters = c("ALK2", "TBILI2"),
+      set_values_to = exprs(
+        AVALC = if_else(AVALC.TBILI2 == "Y" & AVALC.ALK2 == "Y", "Y", "N"),
+        ADTM = pmax(ADTM.TBILI2, ADTM.ALK2),
+        ADTF = if_else(ADTM == ADTM.TBILI2, ADTF.TBILI2, ADTF.ALK2),
+        PARAMCD = "TB2AK2",
+        PARAM = "TBILI > 2 times ULN and ALKPH <= 2 times ULN"
+      ),
+      keep_nas = TRUE
+    ),
+    keys = c("USUBJID")
+  )
+})
```

Comment:
could we move the function out of the expect test?  feel likes it makes it easier if we have to come back and debug/fix something with it separated out

Reply:
Updated